core/src/operations/mod.rs

```diff
@@ -441,274 +552,21 @@ pub enum Operation {
-            Self::And           => 0b0010_0100,
-            Self::Or            => 0b0010_0101,
-            Self::U32and        => 0b0010_0110,
-            Self::U32xor        => 0b0010_0111,
-            Self::FriE2F4       => 0b0010_1000,
-            Self::Drop          => 0b0010_1001,
-            Self::CSwap         => 0b0010_1010,
-            Self::CSwapW        => 0b0010_1011,
-            Self::MLoadW        => 0b0010_1100,
-            Self::MStore        => 0b0010_1101,
-            Self::MStoreW       => 0b0010_1110,
-            Self::FmpUpdate     => 0b0010_1111,
-
-            Self::Pad           => 0b0011_0000,
-            Self::Dup0          => 0b0011_0001,
-            Self::Dup1          => 0b0011_0010,
-            Self::Dup2          => 0b0011_0011,
-            Self::Dup3          => 0b0011_0100,
-            Self::Dup4          => 0b0011_0101,
-            Self::Dup5          => 0b0011_0110,
-            Self::Dup6          => 0b0011_0111,
-            Self::Dup7          => 0b0011_1000,
-            Self::Dup9          => 0b0011_1001,
-            Self::Dup11         => 0b0011_1010,
-            Self::Dup13         => 0b0011_1011,
-            Self::Dup15         => 0b0011_1100,
-            Self::AdvPop        => 0b0011_1101,
-            Self::SDepth        => 0b0011_1110,
-            Self::Clk           => 0b0011_1111,
-
-            Self::U32add        => 0b0100_0000,
-            Self::U32sub        => 0b0100_0010,
-            Self::U32mul        => 0b0100_0100,
-            Self::U32div        => 0b0100_0110,
-            Self::U32split      => 0b0100_1000,
-            Self::U32assert2(_) => 0b0100_1010,
-            Self::U32add3       => 0b0100_1100,
-            Self::U32madd       => 0b0100_1110,
-
-            Self::HPerm         => 0b0101_0000,
-            Self::MpVerify(_)   => 0b0101_0001,
-            Self::Pipe          => 0b0101_0010,
-            Self::MStream       => 0b0101_0011,
-            Self::Split         => 0b0101_0100,
-            Self::Loop          => 0b0101_0101,
-            Self::Span          => 0b0101_0110,
-            Self::Join          => 0b0101_0111,
-            Self::Dyn           => 0b0101_1000,
-            Self::RCombBase     => 0b0101_1001,
-            // <empty>          => 0b0101_1010,
-            // <empty>          => 0b0101_1011,
-            // <empty>          => 0b0101_1100,
-            // <empty>          => 0b0101_1101,
-            // <empty>          => 0b0101_1110,
-            // <empty>          => 0b0101_1111,
-
-            Self::MrUpdate      => 0b0110_0000,
-            Self::Push(_)       => 0b0110_0100,
-            Self::SysCall       => 0b0110_1000,
-            Self::Call          => 0b0110_1100,
-            Self::End           => 0b0111_0000,
-            Self::Repeat        => 0b0111_0100,
-            Self::Respan        => 0b0111_1000,
-            Self::Halt          => 0b0111_1100,
-        }
+    pub fn op_code(&self) -> u8 {
```

Comment:
Should we just get rid of this function? When we need a `u8` value, we should be able to do just `opcode as u8` no?

Reply:
I don't think that this will work: `Operation` enum in our case have variants with fields, so AFAIK this won't allow us to use `as` expression here.

Reply:
Correct, this is considered a "non-primitive cast" and hence `as` doesn't work. This is why I had to use [ToPrimitive](https://docs.rs/num-traits/0.2.19/num_traits/cast/trait.ToPrimitive.html) with `EncodedDecoratorVariant`, but this trait only works for fieldless enums.